### PR TITLE
Fix the calculation of the next payday time in `GetTokenomicsInfo`

### LINKED ISSUE #1240: Reported next payday time is off by one epoch
**Bug Description**

After the P6->P7 protocol update, the next payday time reported by the `GetTokenomicsInfo` endpoint (used by `concordium-client raw GetRewardStatus`) is incorrect. The time shown is one epoch later than the actual payday time.

This is due to the first epoch after the protocol update having effectively 0 duration (it is treated as part of the last epoch before the protocol update).

**Steps to Reproduce**

```
concordium-client --grpc-ip grpc.stagenet.concordium.com --secure raw GetRewardStatus 6c44029eeb721106658d95b3bd1b7761d29451a1030d1d5c583e139a68462340
```
**Expected Result**

```
{
    "bakingRewardAccount": "4",
    "finalizationRewardAccount": "1",
    "foundationTransactionRewards": "0",
    "gasAccount": "3",
    "nextPaydayMintRate": 1.582e-5,
    "nextPaydayTime": "2024-09-18T14:12:10.556Z",
    "protocolVersion": 7,
    "totalAmount": "16079077824676258",
    "totalEncryptedAmount": "111110000000",
    "totalStakedCapital": "5324079444600820"
}
```

**Actual Result**

```
{
    "bakingRewardAccount": "4",
    "finalizationRewardAccount": "1",
    "foundationTransactionRewards": "0",
    "gasAccount": "3",
    "nextPaydayMintRate": 1.582e-5,
    "nextPaydayTime": "2024-09-18T14:15:10.556Z",
    "protocolVersion": 7,
    "totalAmount": "16079077824676258",
    "totalEncryptedAmount": "111110000000",
    "totalStakedCapital": "5324079444600820"
}
```

**Versions**
- Software Version: Node 7.0.3
- OS: all

### PR DESCRIPTION
**Pull request by td202**
## Purpose

Closes #1240

This fixes the timing information for the next payday returned by the `GetTokenomicsInfo` query. Instead of basing it on the genesis block time, the time is computed based on the next epoch transition time. This is more accurate, since after the P6->P7 update, the first epoch (epoch 0) is already considered complete.

## Changes

- Revise the computation of the next payday time, as described above.
- Bump node version for new release.

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [X] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [X] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-consensus/src/Concordium/Queries.hs [hunk 1]
```diff
@@ -920,11 +920,20 @@ getRewardStatus =
...
+            bState <- blockState bp
+            reward <- BS.getRewardStatus bState
+            -- The reward status includes the next payday epoch. To convert this to a UTCTime,
+            -- we get the current epoch and the trigger block time (which we treat as the time of
+            -- the start of the next epoch) from the seed state. For each epoch after the next
+            -- epoch, we add the epoch duration to the trigger block time to get the time of the
+            -- payday.
+            ss <- BS.getSeedState bState
+            let nextEpoch = ss ^. epoch + 1
+            let deltaEpochs e = if e > nextEpoch then fromIntegral (e - nextEpoch) else 0
```
Comment (drsk0) [inline concordium-consensus/src/Concordium/Queries.hs:932]: maybe use `max` here instead `if/then/else`.

**Review by drsk0 — APPROVED** [review]
Looks good, thanks!

**Review by limemloh — APPROVED** [review]

File: concordium-consensus/src/Concordium/Queries.hs (hunk 1 above)
Comment (td202) [reply concordium-consensus/src/Concordium/Queries.hs:932]: `max` is a problem, since the epoch is unsigned.

Comment (drsk0) [reply concordium-consensus/src/Concordium/Queries.hs:932]: I was thinking of `max(fromIntegral(e - nextEpoch), 0)`.

**Review by eb-concordium — APPROVED** [review]

**Pull request merged**